src/io/platform/f3xx/f302x8/UARTf302x8.cpp

```diff
@@ -152,7 +152,7 @@ void UARTf302x8::puts(const char* s) {
 
 char UARTf302x8::getc() {
     uint8_t c;
-    HAL_UART_Receive(&halUART, &c, 1, DEFAULT_TIMEOUT);
+    while(HAL_UART_Receive(&halUART, &c, 1, DEFAULT_TIMEOUT) == HAL_TIMEOUT);
```

Comment:
The DEFAULT_TIMEOUT is pretty low.  It doesn't make much difference, but do we want to increase the timeout to this getc call as well?

Reply:
I think the best approach will be to later make this a parameter with a default option so that users can specify what it is. Since right now this is blocking anyway I think we are fine having a smaller timeout.